include/ptp_clock.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _PTP_CLOCK_H_
+#define _PTP_CLOCK_H_
+
+#include <stdint.h>
+#include <device.h>
+#include <misc/util.h>
+#include <net/ptp_time.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Name of the PTP clock driver */
+#if !defined(PTP_CLOCK_NAME)
+#define PTP_CLOCK_NAME "PTP_CLOCK"
+#endif
+
+struct ptp_clock_driver_api {
+	int (*set)(struct device *dev, struct net_ptp_time *tm);
+	int (*get)(struct device *dev, struct net_ptp_time *tm);
+	int (*adjust)(struct device *dev, int increment);
+	int (*rate_adjust)(struct device *dev, float ratio);
+};
+
+/**
+ * @brief Set the time of the PTP clock.
+ *
+ * @param dev PTP clock device
+ * @param tm Time to set
+ *
+ * @return 0 if ok, <0 if error
+ */
+static inline int ptp_clock_set(struct device *dev, struct net_ptp_time *tm)
+{
+	const struct ptp_clock_driver_api *api = dev->driver_api;
+
+	return api->set(dev, tm);
+}
+
+/**
+ * @brief Get the time of the PTP clock.
+ *
+ * @param dev PTP clock device
+ * @param tm Where to store the current time.
+ *
+ * @return 0 if ok, <0 if error
+ */
+static inline int ptp_clock_get(struct device *dev, struct net_ptp_time *tm)
+{
+	const struct ptp_clock_driver_api *api = dev->driver_api;
+
+	return api->get(dev, tm);
+}
+
+/**
+ * @brief Adjust the PTP clock time.
+ *
+ * @param dev PTP clock device
+ * @param increment Increment of the clock in nanoseconds
+ *
+ * @return 0 if ok, <0 if error
+ */
+static inline int ptp_clock_adjust(struct device *dev, int increment)
+{
+	const struct ptp_clock_driver_api *api = dev->driver_api;
+
+	return api->adjust(dev, increment);
+}
+
+/**
+ * @brief Adjust the PTP clock time change rate when compared to its neighbor.
+ *
+ * @param dev PTP clock device
+ * @param rate Rate the the clock time change
+ *
+ * @return 0 if ok, <0 if error
+ */
+static inline int ptp_clock_rate_adjust(struct device *dev, float rate)
+{
+	const struct ptp_clock_driver_api *api = dev->driver_api;
+
+	return api->rate_adjust(dev, rate);
+}
+
+/**
+ * @brief Return PTP clock that is related to this device.
+ *
+ * @param dev Ethernet device
+ *
+ * @return PTP clock device if found, NULL otherwise.
+ */
+struct device *ptp_clock_lookup_by_dev(struct device *dev);
```

Comment:
you don't need that function afaik, net_eth_get_ptp_clock() does what you want already

Reply:
I would need it in gptp, which actually uses "struct device" instead of iface, thus I kept the function as is.

Reply:
why gptp, a network code, uses device directly?

Reply:
I could use iface in gptp code instead of directly storing device so this can be changed actually.